### PR TITLE
feat(nimbus): Firefox Labs Model and API changes

### LINKED ISSUE #11556: Model and API changes for Firefox Labs Integration
The NimbusExperiment model needs to be updated to add fields for the schema changes in phase 1. These fields need to be added to the v6 API and the v8 API.

No API or UI changes need to be made for this. The relevant fields will have to be made available in the Django admin panel for editing. (The Nimbus front-end is currently undergoing a rewrite and any new code for the old UI would be thrown out soon, resulting in wasted effort. Instead, we can live with a Django admin panel solution until the new UI is released since we expect low throughput of Firefox Labs opt-ins.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-4877)

### PR DESCRIPTION
**Pull request by Herraj**
Because

- We want to add the fields from the updated `NimbusExperiment` schema to the `NimbusExperiment` model to be available internally. 

This commit

- Updates `NimbusExperiment` model.
- Updates `NimbusBranchSerializerDesktop` model.
- Updates `v6` and `v8` serializer classes with the above fields.
- Generates new db migration for the above fields.

Fixes #11556

### CODE REVIEW

File: experimenter/experimenter/experiments/models.py [hunk 1]
```diff
@@ -370,6 +370,13 @@ class NimbusExperiment(NimbusConstants, TargetingConstants, FilterMixin, models.
     )
     use_group_id = models.BooleanField(default=False)
     objects = NimbusExperimentManager()
+    is_firefox_labs_opt_in = models.BooleanField("Is Firefox Labs Opt In", default=False)
```
Comment (brennie) [inline experimenter/experimenter/experiments/models.py:373]: "Is this experiment a Firefox Labs opt-in?"

File: experimenter/experimenter/experiments/models.py [hunk 2]
```diff
@@ -370,6 +370,13 @@ class NimbusExperiment(NimbusConstants, TargetingConstants, FilterMixin, models.
...
+    firefox_labs_title = models.TextField(
+        "Firefox Labs experiment title", blank=True, null=True
+    )
+    firefox_labs_description = models.TextField(
+        "Firefox Labs experiment description", blank=True, null=True
```
Comment (brennie) [inline experimenter/experimenter/experiments/models.py:378]: Please mention that these are fluent string IDs.

File: experimenter/experimenter/experiments/models.py [hunk 3]
```diff
@@ -1345,6 +1352,9 @@ class NimbusBranch(models.Model):
     slug = models.SlugField(max_length=NimbusConstants.MAX_SLUG_LEN, null=False)
     description = models.TextField(blank=True, default="")
     ratio = models.PositiveIntegerField(default=1)
+    firefox_labs_title = models.TextField(
```
Comment (brennie) [inline experimenter/experimenter/experiments/models.py:1355]: please mention this is a fluent string ID

File: experimenter/experimenter/experiments/tests/factories.py [hunk 4]
```diff
@@ -468,6 +468,9 @@ class NimbusExperimentFactory(factory.django.DjangoModelFactory):
     qa_status = factory.LazyAttribute(
         lambda o: random.choice(list(NimbusExperiment.QAStatus)).value
     )
+    is_firefox_labs_opt_in = factory.LazyAttribute(lambda o: False)
+    firefox_labs_title = factory.LazyAttribute(lambda o: faker.text(200))
+    firefox_labs_description = factory.LazyAttribute(lambda o: faker.text(200))
```
Comment (brennie) [inline experimenter/experimenter/experiments/tests/factories.py:473]: 200 characters is huge for what goes in these fields. lets do `slugify(faker.catchphrase()`

File: experimenter/experimenter/experiments/tests/factories.py [hunk 5]
```diff
@@ -704,6 +707,7 @@ class NimbusBranchFactory(factory.django.DjangoModelFactory):
         lambda o: slugify(o.name)[: NimbusExperiment.MAX_SLUG_LEN]
     )
     description = factory.LazyAttribute(lambda o: faker.text())
+    firefox_labs_title = factory.LazyAttribute(lambda o: faker.text())
```
Comment (brennie) [inline experimenter/experimenter/experiments/tests/factories.py:710]: same here re slugify(faker.catchphrase())

File: experimenter/experimenter/experiments/tests/api/v8/test_serializers.py [hunk 6]
```diff
@@ -94,6 +97,9 @@ def test_expected_schema_with_desktop(self):
                 "localizations": None,
                 "locales": ["en-US"],
                 "publishedDate": experiment.published_date,
+                "isFirefoxLabsOptIn": False,
+                "firefoxLabsTitle": None,
+                "firefoxLabsDescription": None,
             },
         )
```
Comment (yashikakhurana) [inline experimenter/experimenter/experiments/tests/api/v8/test_serializers.py:105]: Would also suggest to add a test for the non-default values

Comment (Herraj) [reply experimenter/experimenter/experiments/tests/api/v8/test_serializers.py:105]: hmm yep I agree

**Pull request merged**